spec.bs

```diff
@@ -1738,6 +1754,52 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
   </wpt>
 </div>
 
+<div algorithm>
+  The <dfn method for=Fence>disableUntrustedNetwork()</dfn> method steps are:
+
+  1. Let |p| be [=a new promise=].
+
+  1. Let |instance| be [=this=]'s [=relevant global object=]'s [=Window/browsing context=]'s
+     [=browsing context/fenced frame config instance=].
+
+  1. If |instance| is null, then [=resolve=] |p| with {{undefined}} and return |p|.
+
+  1. If |instance|'s [=fenced frame config instance/can disable untrusted network=] is false, then
+     [=resolve=] |p| with {{undefined}} and return |p|.
+
+  1. If the [=relevant settings object=]'s [=environment settings object/origin=] and |instance|'s
+     [=fenced frame config instance/mapped url=]'s [=url/origin=] are not [=same origin=], then
+     [=resolve=] |p| with {{undefined}} and return |p|.
```

Comment:
Copied from [this comment](https://github.com/WICG/fenced-frame/pull/176#discussion_r1748934441):

> Let's very briefly describe in a note what this condition means. I think it means that the fenced frame has navigated away from the origin described by its original config (?), but in that case I would expect the config to be null anyways. Maybe I'm wrong about that part though.

Reply:
I'm also expecting that to be the case. I think this check is redundant. If we want to be extra safe (and make it clear to implementers that only documents that are same-origin to the mapped URL can call this), we can assert they're same origin instead. But let me know if you think even that's unnecessary.